File: src/ProjectR.Scripting/content/scripts/spells/lua/Netherball.lua
function GetTargetType()
    return Single
end

function GetName()
    return "Netherball"
end

function GetDescription()
    return "Shoot a weak Netherball at your enemies. Recovers some MP."
end

function IsSupportSpell()
    return false
end

function GetMPCost()
    return 0
end

function GetDelay()
  return .5
end

function GetSpellType()
  return Pure
end

function GetMasteries()
  return {  }
end

function SpellEffect()
    d:TakeDamage(aAD + aMD)
    a:UseMP(-(aMP * .2))
end

File: src/ProjectR.Scripting/content/scripts/spells/Netherball.cs
using System.Collections.Generic;
using ProjectR.Interfaces;
using ProjectR.Interfaces.Model;
using ProjectR.Interfaces.Model.Stats;

namespace ProjectR.Scripting
{
    public class Netherball : SpellScriptBase
    {
        public override string Name { get { return "Netherball"; } }
        public override string Description { get { return "Shoot a weak Netherball at your enemies. Recovers some MP."; } }

        public override TargetType TargetType { get { return TargetType.Single; } }
        public override IList<EleMastery> Masteries { get { return new EleMastery[] { }; } }
        public override SpellType SpellType { get { return SpellType.Pure; } }
        public override bool IsSupportSpell { get { return false; } }

        public override double MPCost { get { return 0; } }
        public override double Delay { get { return .5; } }

        public override void SpellEffect(ICharacter caster, ICharacter target)
        {
            DealDamage(cAD + cMD);
            caster.UseMP(-(cMP * .2));
        }
    }
}